wordpress-mobile/WordPress-iOS
# [SIWA] Handle invalid Apple user

Fixes #12399 
Related WPAuth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/137

This uses the WPAuth changes to:
- Save the Apple user id to the keychain.
- On app launch, get the user id from the keychain and check it's state.
- If the user id is no longer authorized, log out of the account (using the same logic as logging out from the Me tab).

To note, this only addresses the scenario in the testing steps, i.e. checking state on app launch. We're not yet actively listening for Apple ID disconnects (https://github.com/wordpress-mobile/WordPress-iOS/issues/12535). Stay tuned!

To test:

This should be tested on a real device for proper keychain handling.

- Log in / sign up with Apple.
- Disconnect in your Apple account on the device.
  - Go to iOS Settings.
  - Select your iCloud account at the top.
  - Go to Password & Security > Apps Using Your Apple ID.
  - Select WordPress > Stop Using Apple ID.
- Rerun the app.
- The app should log you out.
- This message should appear in the logs: `checkAppleIDCredentialState: Unauthorized Apple ID. User signed out.`

Update release notes:
